src/components/ForecastList.js

```diff
@@ -1,40 +1,92 @@
 import React, { Component } from 'react';
-import { View, FlatList, Text } from 'react-native';
-import * as weatherData from '../services/weatherData';
+import { View, FlatList, Text, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import { autobind } from 'core-decorators';
+import ForecastListItem from './ForecastListItem';
+import * as weatherService from '../services/weather';
 
 class ForecastList extends Component {
+  static propTypes = {
+    city: PropTypes.object,
+    weatherReport: PropTypes.object,
+  };
+
   constructor(props) {
     super(props);
 
     this.state = {
-      list: [],
+      forecastByDaysList: [],
     };
   }
 
   componentDidMount() {
-    weatherData.fetchWeatherForcastForCity('Zagreb', this.fetchWeather.bind(this));
+    const {
+      city: { value },
+    } = this.props;
+
+    weatherService.fetchWeatherForcastForCity(value, this.fetchWeather);
   }
 
+  @autobind
   fetchWeather(weather) {
     this.setState({
-      list: weather,
+      forecastByDaysList: weather,
     });
   }
 
+  renderItem = ({ item, index }) => <ForecastListItem forecast={item} key={`${index}`} />;
```

Comment:
`@autobind` and "normal" function (not arrow function). Also, do we have something else from the `item` itself that could serve as a key (instead of `index`)?